script.js

```diff
@@ -52,4 +52,112 @@ function valutaConverter(event) {
   }
 }
 
-// updateing existing currency conversion rates with new rates form----------------
+// updateing existing currency conversion rates with new rates form
+// **********************************************
+// Week2 / An aaray of the currency rate objects
+// **********************************************
+
+const brandObjectArray = [
+  {
+    timestamp: Date.now(),
+    base: "EUR",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+  {
+    timestamp: Date.now(),
+    base: "USD",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+  {
+    timestamp: Date.now(),
+    base: "DKK",
+    date: new Date().toLocaleDateString(),
+    rates: {},
+  },
+];
+
+document
+  .getElementById("brand-currency-form-beta")
+  .addEventListener("submit", brandCurrencyBeta);
+
+function brandCurrencyBeta(event) {
+  event.preventDefault();
+
+  //getting and filling objects of array------------
+  const baseCurrency = document.getElementById("base-currency-beta").value;
+  const quoteName = document.getElementById("quote-name-beta").value;
+  const quoteRate = document.getElementById("quote-rate-beta").value;
+
+  if (baseCurrency === "EUR") {
+    brandObjectArray[0].rates[quoteName] = +quoteRate;
+  } else if (baseCurrency === "USD") {
+    brandObjectArray[1].rates[quoteName] = +quoteRate;
+  } else if (baseCurrency === "DKK") {
+    brandObjectArray[2].rates[quoteName] = +quoteRate;
+  }
+  document.getElementById("quote-rate-beta").value = "";
+  console.log(brandObjectArray);
+
+  //Calling function: Display all base currencies and their rates
+  displayBaseAndQuoteCurrenciesAndRates();
```

Comment:
this function only loops through the rates and log them into console, right? If so, you can delete it, it's not used by the app. The user should not! see anything in the console. it is only for developers

Reply:
Yes, that is true.
Ok, thank you. I thought that we were also supposed to show them.
I have deleted
//Calling function: Display all base currencies and their rates
  displayBaseAndQuoteCurrenciesAndRates();
  
  //Function: checking base currency and showing existing quotes and rate for each base ---------
function displayBaseAndQuoteCurrenciesAndRates() {
  console.log("Displaying the Base and Quote currencies and conversion rates:");
  for (currency of brandObjectArray) {
    if (currency.base === "EUR") {
      console.log("EUR base:");
      console.log(currency.rates);
    } else if (currency.base === "USD") {
      console.log("USD base:");
      console.log(currency.rates);
    } else {
      console.log("DKK base:");
      console.log(currency.rates);
      //console.log(`${currency.rates}`);  Why this ${currency.rates} doesn't work?! It gives [Object Object]
    }
  }
}